---
title: "editing_csv_files"
author: "Alexa Byers"
date: "2025-01-12"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
library(ggplot2)
library(reshape2)
library(dplyr)
#install.packages("FSA")
library(FSA)
#install.packages("ggstatsplot")
library(ggstatsplot)
```


## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

#editing the csv files of the gene coverage to retain only those found in the genome annotation files

#Gene coverage dataset: METABOLIC annotation which will be used in analysis of biogeochemical cycling genes
#Contig coverage values were normalised by read depth of sample and scaled by average library size
#this is the output from the normalise_contig_coverage.R script
```{r}
contig_coverage <- read.csv("C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/Gene coverage/normalised_coverage_METABOLIC.csv") 
rownames(contig_coverage) <- contig_coverage$contigName 
contig_coverage <- contig_coverage[-c(1:2)]
#edit colnames to reflect sample file ID
names(contig_coverage) = gsub(pattern = ".bam", replacement = "", x = names(contig_coverage))
```
#METABOLIC ANNOTATION DATASET
```{r}
contig_functions_METABOLIC <- read.csv("C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/METABOLIC/METABOLIC_annotations_refined.csv")
rownames(contig_functions_METABOLIC) <- contig_functions_METABOLIC$Contig
contig_functions_METABOLIC <- contig_functions_METABOLIC[,-1]
```
```{r}
nrow(contig_coverage) #457097 contigs
nrow(contig_functions_METABOLIC)#6349 contigs annotated by METABOLIC
(6349/457097)*100 #1.39% of protein coding regions contigs were annotated by METABOLIC as biogeochemical cycling genes
```
#filter gene coverage file to retain hits found in METABOLIC annotation
```{r}
contigs_keep_METABOLIC <- rownames(contig_functions_METABOLIC)
contig_coverage_METABOLIC <- contig_coverage[rownames(contig_coverage) %in% contigs_keep_METABOLIC ,]
write.csv(contig_coverage_METABOLIC, "C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/Gene coverage/gene_coverage_METABOLIC_filtered.csv")
```
##subset DRAM annotation file to keep only contigs that could not be annotated by METABOLIC
```{r read in DRAM annotations dataset}
DRAM_annotations <- read.csv("C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/DRAM/DRAM_annotations.csv")
rownames(DRAM_annotations) <- DRAM_annotations$Contig_ID
```
```{r remove contigs already annoted by METABOLIC}
contigs_remove_DRAM <- rownames(contig_functions_METABOLIC)
DRAM_annotations_filtered <- DRAM_annotations[ ! rownames(DRAM_annotations) %in% contigs_remove_DRAM ,]
nrow(contig_functions_METABOLIC) #6349 contigs annotated by METABOLIC
nrow(DRAM_annotations_filtered) #13002 contigs annotated by DRAM
```
```{r}
DRAM_annotations_filtered$Gene_ID <- as.factor(DRAM_annotations_filtered$Gene_ID)
levels(DRAM_annotations_filtered$Gene_ID)
write.csv(DRAM_annotations_filtered, "C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/DRAM/DRAM_annotations_filtered.csv")
```

##COMBINED METABOLIC AND DRAM ANNOTATIONS IN EXCEL. THIS IS THE SPREADSHEET I WILL USE 
##IN THE MAIN ANALYSIS, HOWEVER MAKE IT CLEAR THEY ARE COMING FROM TWO ANNOTATION DATASETS
```{r}
contig_functions_MET_DRAM <- read.csv("C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/phyloseq/METABOLIC_DRAM_combined.csv")
rownames(contig_functions_MET_DRAM) <- contig_functions_MET_DRAM$Contig
contig_functions_MET_DRAM <- contig_functions_MET_DRAM[,-1]
```
```{r}
nrow(contig_coverage) #467097 contigs
nrow(contig_functions_MET_DRAM)#23512 contigs annotated by METABOLIC and DRAM
```
```{r}
contigs_keep_MET_DRAM <- rownames(contig_functions_MET_DRAM)
contig_coverage_MET_DRAM <- contig_coverage[rownames(contig_coverage) %in% contigs_keep_MET_DRAM ,]
write.csv(contig_coverage_MET_DRAM, "C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/Gene coverage/gene_coverage_MET_DRAM_filtered.csv")
write.csv(contig_coverage_MET_DRAM, "C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/phyloseq/gene_coverage_MET_DRAM_filtered.csv")
```

## 20th Jan edits filtering DRAM energy annotations 
```{r}
DRAM_annotations <- read.csv("C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/DRAM/DRAM_annotations_ENERGY.csv")
rownames(DRAM_annotations) <- DRAM_annotations$Contig_ID
```
```{r remove contigs already annoted by METABOLIC and DRAM file}
contigs_remove_DRAM <- rownames(contig_functions_MET_DRAM)
DRAM_annotations_filtered <- DRAM_annotations[ ! rownames(DRAM_annotations) %in% contigs_remove_DRAM ,]
(nrow(DRAM_annotations_filtered)) - (nrow(DRAM_annotations)) #4103 additional contigs annotated
#write.csv(DRAM_annotations_filtered, "C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/DRAM/energy_annotations_FILT.csv")
```
```{r subset copy of contig_functions_MET_DRAM select KO numbers with annotations}
DRAM_annotations_filtered$gene_id <- as.factor(DRAM_annotations_filtered$gene_id)
KO_to_KEEP <- levels(DRAM_annotations_filtered$gene_id)
contig_functions_MET_DRAM_copy <- contig_functions_MET_DRAM
contig_functions_MET_DRAM_copy2 <- contig_functions_MET_DRAM_copy[contig_functions_MET_DRAM_copy$KO_number %in% KO_to_KEEP ,]
#write.csv(contig_functions_MET_DRAM_copy2, "C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/DRAM/energy_annotations_DONE.csv")
```
```{r subset copy of DRAM_annotations_filtered without METABOLIC annotations}
contig_functions_MET_DRAM$KO_number <- as.factor(contig_functions_MET_DRAM$KO_number)
KO_to_REMOVE <- levels(contig_functions_MET_DRAM$KO_number)

DRAM_annotations_filtered2 <- DRAM_annotations_filtered
DRAM_annotations_filtered3 <- DRAM_annotations_filtered2[ ! DRAM_annotations_filtered2$gene_id %in% KO_to_REMOVE ,]
#write.csv(DRAM_annotations_filtered3, "C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/DRAM/energy_annotations_TODO.csv")

DRAM_annotations_filtered4 <- DRAM_annotations_filtered
DRAM_annotations_filtered5 <- DRAM_annotations_filtered4[ DRAM_annotations_filtered4$gene_id %in% KO_to_REMOVE ,]
#write.csv(DRAM_annotations_filtered5, "C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/DRAM/energy_annotations_completed.csv")
```
##count number of contigs in each sample
```{r}
total_contigs <- as.data.frame(colSums(contig_coverage != 0))
METABOLIC_DRAM_contigs <- as.data.frame(colSums(contig_coverage_MET_DRAM != 0))
contigs_annotated <- cbind(total_contigs, METABOLIC_DRAM_contigs)
write.csv(contigs_annotated, "C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/phyloseq/number_contigs_annotated.csv")
```
##PROCESSING CAZy ANNOTATIONS
```{r}
contig_functions_CAZy <- read.csv("C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/dbCAN/dbCAN.csv")
rownames(contig_functions_CAZy) <- contig_functions_CAZy$Gene.ID
```
#filter gene coverage file to retain hits found in METABOLIC annotation
```{r}
contigs_keep_CAZy <- rownames(contig_functions_CAZy)
contig_coverage_CAZy <- contig_coverage[rownames(contig_coverage) %in% contigs_keep_CAZy ,]
write.csv(contig_coverage_CAZy, "C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/Gene coverage/gene_coverage_CAZy_filtered.csv")
```
```{r}
total_contigs <- as.data.frame(colSums(contig_coverage != 0))
CAZy_contigs <- as.data.frame(colSums(contig_coverage_CAZy != 0))
contigs_annotated <- cbind(total_contigs, CAZy_contigs)
write.csv(contigs_annotated, "C:/Users/byersa/OneDrive - Lincoln University/Documents/Rutherford/Metagenomics/phyloseq/contigs_annotated_CAZy.csv")
```


